pyproject.toml

```
@@ -75,6 +75,27 @@ Changelog = "https://cotainr.readthedocs.io/en/stable/release_notes/index.html"
 [tool.hatch.version]
 path = "cotainr/__init__.py"
 
+[tool.hatch.build.targets.sdist]
+include = [
+    "/cotainr",
+    "/doc",
+    "README.md",
+    "LICENSE",
+    "pyproject.toml",
+    ".pre-commit-config.yaml",
+    ".readthedocs.yml",
+]
+exclude = [
+    "/examples",
+    "/bin"
+]
+
```

We might want to also exclude the test suite from the wheel:
```suggestion

[tool.hatch.build.targets.wheel]
exclude = [
    "/cotainr/tests"
]

```

---

I believe the [tool.hatch.build.targets.wheel] table does not have such straightforward notation: https://hatch.pypa.io/1.9/config/build/#artifacts. I can try to configure to achieve a similar effect, however I believe the "correct" way is probably to change to the [standard python package folder structure with /src and /tests](https://packaging.python.org/en/latest/tutorials/packaging-projects/#creating-the-package-files)

---

I think the [patterns notation](https://hatch.pypa.io/1.13/config/build/#patterns) is the same for all targets. At least my suggested code worked when I tried it locally. Artifacts seems to be more about including stuff that is ignored by git.

---

Good to know! I did not test it previously, but yes it does seem to work as expected.

---

See https://github.com/DeiC-HPC/cotainr/pull/120#issuecomment-2729949820 too :)

---

With the config in this PR, the wheel should now be clean and only contain the actual cotainr code. Only the sdist archive contains more pieces.